src/storage/storage.cc

```diff
@@ -243,7 +243,7 @@ Status Storage::CreateColumnFamilies(const rocksdb::Options &options) {
   return Status::OK();
 }
 
-Status Storage::Open(bool read_only) {
+Status Storage::Open(bool read_only, bool as_secondary) {
```

Comment:
You can add a `enum OpenMode`, then we can have 3 state here rather than 2*2=4.

Reply:
Sure, however, I also noticed an issue [here](https://github.com/apache/kvrocks/blob/unstable/src/storage/storage.cc#L329). The check for `s` doesn't make sense; it doesn't actually validate the return value of the DBOpen operation. I can fix this issue together.